engines/terraform/terraform.go

```diff
@@ -6,11 +6,40 @@ import (
 	"fmt"
 	"io"
 	"path/filepath"
+	"runtime/debug"
 
 	app_spec_schema "github.com/nitrictech/suga/cli/pkg/schema"
 	"github.com/nitrictech/suga/engines"
 )
 
+// PanicError represents an error that occurred due to a panic during terraform operations
+type PanicError struct {
+	OriginalPanic interface{}
+	StackTrace    []byte
+	LogFilePath   string
+	Target        string
+}
+
+func (e *PanicError) Error() string {
+	if e.LogFilePath != "" {
+		return fmt.Sprintf("build panic occurred for target '%s' - details logged to %s", e.Target, e.LogFilePath)
+	}
+	return fmt.Sprintf("build panic occurred: %v", e.OriginalPanic)
+}
```

Comment:
This statefulness feels a bit odd, could the state of whether or not the error was logged be managed at the point of whether or not we know its been written to a file or not?